packages/dataviews/src/components/dataviews-view-config/index.tsx

```diff
@@ -8,143 +8,216 @@ import type { ChangeEvent } from 'react';
 	return (
 		<DropdownMenu
 			trigger={
-				<DropdownMenuItem
-					suffix={ <span aria-hidden="true">{ view.perPage }</span> }
-				>
-					<DropdownMenuItemLabel>
-						{ __( 'Items per page' ) }
-					</DropdownMenuItemLabel>
-				</DropdownMenuItem>
+				<Button
+					size="compact"
+					icon={ activeView?.icon }
+					label={ __( 'Layout' ) }
+				/>
 			}
 		>
-			{ PAGE_SIZE_VALUES.map( ( size ) => {
+			{ availableLayouts.map( ( layout ) => {
+				const config = VIEW_LAYOUTS.find( ( v ) => v.type === layout );
+				if ( ! config ) {
+					return null;
+				}
 				return (
 					<DropdownMenuRadioItem
-						key={ size }
-						value={ size }
-						name="view-actions-page-size"
-						checked={ view.perPage === size }
-						onChange={ () => {
-							onChangeView( {
-								...view,
-								// `e.target.value` holds the same value as `size` but as a string,
-								// so we use `size` directly to avoid parsing to int.
-								perPage: size,
-								page: 1,
-							} );
+						key={ layout }
+						value={ layout }
+						name="view-actions-available-view"
+						checked={ layout === view.type }
+						hideOnClick
+						onChange={ ( e: ChangeEvent< HTMLInputElement > ) => {
+							switch ( e.target.value ) {
+								case 'list':
+								case 'grid':
+								case 'table':
+									return onChangeView( {
+										...view,
+										type: e.target.value,
+										...defaultLayouts[ e.target.value ],
+									} );
+							}
+							warning( 'Invalid dataview' );
 						} }
 					>
-						<DropdownMenuItemLabel>{ size }</DropdownMenuItemLabel>
+						<DropdownMenuItemLabel>
+							{ config.label }
+						</DropdownMenuItemLabel>
 					</DropdownMenuRadioItem>
 				);
 			} ) }
 		</DropdownMenu>
 	);
 }
 
-function FieldsVisibilityMenu< Item >( {
-	view,
-	onChangeView,
-	fields,
-}: FieldsVisibilityMenuProps< Item > ) {
+interface ViewActionsProps {
+	defaultLayouts?: SupportedLayouts;
+}
+
+function SortFieldControl() {
+	const { view, fields, onChangeView } = useContext( DataViewsContext );
+	const orderOptions = useMemo( () => {
+		const sortableFields = fields.filter(
+			( field ) => field.enableSorting !== false
+		);
+		return sortableFields.map( ( field ) => {
+			return {
+				label: field.label,
+				value: field.id,
+			};
+		} );
+	}, [ fields ] );
+
+	return (
+		<SelectControl
+			__nextHasNoMarginBottom
+			__next40pxDefaultSize
+			label={ __( 'Sort by' ) }
+			value={ view.sort?.field }
+			options={ orderOptions }
+			onChange={ ( value: string ) => {
+				onChangeView( {
+					...view,
+					sort: {
+						direction: view?.sort?.direction || 'desc',
+						field: value,
+					},
+				} );
+			} }
+		/>
+	);
+}
+
+function SortDirectionControl() {
+	const { view, onChangeView } = useContext( DataViewsContext );
+	return (
+		<ToggleGroupControl
+			className="dataviews-view-config__sort-direction"
+			__nextHasNoMarginBottom
+			__next40pxDefaultSize
+			isBlock
+			label={ __( 'Order' ) }
```

Comment:
If I click the "sort by" label, the select is focused properly, so the label is working properly but it's not the case for "Order" or "Items per page". I expect the button group (or the first button) to be focused. It seems we might have an issue in the component. cc @tyxla

Reply:
I have thought about this, but I'm not sure on what basis we would want to implement that behavior. Usually we do it for accessibility (i.e. increase the click target on things like checkboxes and radios, or because "the browser does it by default" when an `input` is properly associated with a `label`). In the case of compound components like ToggleGroupControl, I don't think there's a click target issue, and since in normal HTML semantics the label would actually be associated with a `radiogroup`/`group`, or be a `legend` in a `fieldset`, that kind of click-to-focus behavior would not happen.

So the only reason to add click-label-to-focus on compound components would be for UX reasons (consistency), which may possibly be bad for accessibility due to being non-standard 🤔

Reply:
The behavior here is expected AFAIK, because the underlying`ToggleGroupControl` behaves like a radiogroup with radio buttons. You can see an example with actual radio buttons here: https://ariakit.org/components/radio where you can tab into and out of the component with `Tab` / `Shift` + `Tab`, and you can use the arrows to select a different radio button.

If you're referencing the focus styles, this is another thing: we've been debating this in https://github.com/WordPress/gutenberg/pull/63450#issuecomment-2228354818 and decided to not move the focus to the `ToggleGroupControlOption` because, it a nutshell, it would look odd with the selected style.  We can always re-evaluate that if you feel like it was not the best path forward.

Reply:
> So the only reason to add click-label-to-focus on compound components would be for UX reasons (consistency), which may possibly be bad for accessibility due to being non-standard 🤔

Yes, that's my point for a user, it's a single control, just like the others in the form. It's not really clear why there's a different between these controls and the "sort by" for instance. 

I can live with that for now, but for me it's still a bug and there should be a way to fix the UX without breaking the a11y IMO.